### PR TITLE
Update user guide

### LINKED ISSUE #1794: Update image examples in user guide
Since the removal of the "Lena" image from `scipy`, our examples in the user guide are out of date. The code of the examples has been changed to use `ascent()` instead of `lena()`, but the actual image files still show examples from the old code, which should be updated.
  
See, for example, [here](http://hyperspy.org/hyperspy-doc/current/user_guide/visualisation.html#plotting-several-images).

### PR DESCRIPTION
**Pull request by jat255**
This PR resolves #1794, and makes a number of other changes to the user guide:

Also resolves #1715 

Summary of changes:

* Removed all references and pictures with Lena, replacing them with `scipy.misc.ascent()`
* Updated plot examples so they match with the new default styles of `matplotlib` (so a user should get similar results when running the code themselves)
* Fixed a number of typos, and minor formatting issues throughout the user guide
* Revamped bibliography page to include links to referenced articles
* Shortened all lines in the source files to be under 80 columns
* Added some gif screenshots to demonstrate some interactive visualizations
* Properly added favicon so the logo is shown in the tab title when looking at the user guide
* Probably a few other things...

Note, I didn't get around to automating the plot outputs like discussed in #1794 

The docs are building okay on my machine, but I don't know if there's a way to test it before merging on readthedocs. Anyone know?

### CODE REVIEW

File: doc/dev_guide.rst [hunk 1]
```diff
@@ -146,15 +152,17 @@ You can check your code with the `pep8 Code Checker
 Write tests
 ^^^^^^^^^^^
 
-Every new function that is writen in to HyperSpy needs to be tested and documented.
+Every new function that is writen in to HyperSpy needs to be tested and
```
Comment (magnunor) [inline doc/dev_guide.rst]: writen -> written

File: doc/dev_guide.rst [hunk 2]
```diff
@@ -183,8 +191,11 @@ Useful hints on testing:
   code, which is often error-prone. See `pytest documentation for more details
   <http://doc.pytest.org/en/latest/parametrize.html>`_.
 * It is good to check that the tests does not use too much of memory after
-  creating new tests. If you need to explicitely delete your objects and free memory, you can do the following to release the memory associated to the ``s`` object, for example:
-::
+  creating new tests. If you need to explicitely delete your objects and free
```
Comment (magnunor) [inline doc/dev_guide.rst]: explicitely -> explicitly ?

**Comment by magnunor** [comment]
Good stuff! I skimmed through the changes, but the changes are a bit too numerous to go through in detail.

I especially like the use of gifs to show interactive functionality, which I for some reason haven't thought about before. I'll probably use it to show stuff in Atomap at some point in the future.

**Comment by ericpre** [comment]
Thanks @jat255 for the tedious task...
Just two minor comments:
- would it be worth to format the bibliography in a way the references are aligned vertically? For example, something along the line of the [bilbiography of pytables](http://www.pytables.org/usersguide/bibliography.html).
- tackle #1715 while you are at it? Or @magnunor, do you want to finish it?

**Comment by jat255** [comment]
@ericpre 

> would it be worth to format the bibliography in a way the references are aligned vertically? For example, something along the line of the bilbiography of pytables.

I was wondering what the best way to get this to look nicer would be. I'll implement that.

> tackle #1715 while you are at it?

That's not too tricky either, so I can get that taken care of as well

@magnunor Thanks for pointing out a few more typos. I'm going to go through the spell check of Pycharm on the whole project and see if I can find any more, as well.

**Comment by jat255** [comment]
> would it be worth to format the bibliography in a way the references are aligned vertically? For example, something along the line of the bilbiography of pytables.

> tackle #1715 while you are at it?

Okay, both of these are taken care of.

**Comment by jat255** [comment]
Ok, I think this should be ready for review, unless there are any other changes desired.

**Comment by jat255** [comment]
FYI, how the table of contents looks with adjusted `CHANGES.rst`:

image

And how the new bibliography page looks:

image

**Comment by francisco-dlp** [comment]
LGTM, @ericpre ?

**Comment by ericpre** [comment]
I may be a bit picky there, but are the subheadings "current version" and
"previous versions" necessary ? Other than that it's all good!

**Comment by jat255** [comment]
> I may be a bit picky there, but are the subheadings "current version" and
> "previous versions" necessary ?

It was the easiest way I could find to hide the version numbers, since the TOC is limited to a depth of 3. The other option would be to limit the depth to 2, or reformat the CHANGES file, but both of those didn't sound appropriate. If there's another option, I'd certainly be open to it.

**Comment by ericpre** [comment]
Ok, yes it makes sense!

**Pull request merged**